### PR TITLE
Added_test_cases_moead

### PR DESCRIPTION
**Pull request by Sainithin-bit**
Added BBSMOED and DefaultMOED test cases against ZDT-3 test suit. Previously , only DirichletMOEAD was tested.

### CODE REVIEW

File: tests/moead_test.cpp [hunk 1]
```diff
@@ -628,3 +628,80 @@ TEST_CASE("MOEADDIRICHLETZDT3Test", "[MOEADTest]")
   const arma::cube& finalPopulation = opt.ParetoSet();
   REQUIRE(VariableBoundsCheck(finalPopulation));
 }
+
+
```
Comment (zoq) [inline tests/moead_test.cpp:632]:
```suggestion
```

Remove the extra two empty lines.

File: tests/moead_test.cpp [hunk 2]
```diff
@@ -628,3 +628,80 @@ TEST_CASE("MOEADDIRICHLETZDT3Test", "[MOEADTest]")
...
+  arma::mat coords = ZDT_THREE.GetInitialPoint();
+  std::tuple<ObjectiveTypeA, ObjectiveTypeB> objectives = ZDT_THREE.GetObjectives();
+
+  opt.Optimize(objectives, coords);
+
+  const arma::cube& finalPopulation = opt.ParetoSet();
+  REQUIRE(VariableBoundsCheck(finalPopulation));
+}
+
```
Comment (zoq) [inline tests/moead_test.cpp:670]:
```suggestion
```

Remove the empty line.

**Review by zoq — COMMENTED** [review]
Did you run the test cases in a loop with different seeds to check how often the test passes/fails?

**Comment by Sainithin-bit** [comment]
No , i haven't done that

**Comment by Sainithin-bit** [comment]
Hello @zoq , I know this happened a while back i was engaged in something, but how to run this test you mentioned.

**Comment by Sainithin-bit** [comment]
Never mind , i got it

**Pull request closed without merging**